### PR TITLE
Add subscription scope to assessment metadata API

### PR DESCRIPTION
**Pull request by tarosler**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by tarosler** [comment]
the assessmentMetadata.json file already reviewed and approved [here](https://github.com/chlahav/azure-rest-api-specs/blob/master/specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/assessmentsMetadata.json). This PR is adding two new paths (subscription scope) to the file.

File: specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/assessmentsMetadata.json [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../common/v1/types.json#/definitions/CloudError"
+            }
+          }
+        }
+      }
+    },
+        "/subscriptions/{subscriptionId}/providers/Microsoft.Security/assessmentsMetadata": {
```
Comment (yungezz) [inline specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/assessmentsMetadata.json]: indention

File: specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/examples/AssessmentsMetadata/GetAssessmentsMetadata_example.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        "name": "21300918-b2e3-0346-785f-c77ff57d243b",
+        "type": "Microsoft.Security/assessmentMetadata",
+        "properties": {
+          "displayName": "Install endpoint protection solution on virtual machine scale sets",
+          "policyDefinitionId": "/providers/Microsoft.Authorization/policyDefinitions/26a828e1-e88f-464e-bbb3-c134a282b9de",
+          "description": "Install an endpoint protection solution on your virtual machines scale sets, to protect them from threats and vulnerabilities.",
+          "remediationDescription": "To install an endpoint protection solution: 1.  <a href=\"https://docs.microsoft.com/azure/virtual-machine-scale-sets/virtual-machine-scale-sets-faq#how-do-i-turn-on-antimalware-in-my-virtual-machine-scale-set\">Follow the instructions in How do I turn on antimalware in my virtual machine scale set</a>",
+          "category": [ "Compute" ],
+          "secureScoreWeight": 15,
+		  "assessmentType": "Custom"
```
Comment (yungezz) [inline specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/examples/AssessmentsMetadata/GetAssessmentsMetadata_example.json]: idention

File: specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/examples/AssessmentsMetadata/ListAssessmentsMetadata_example.json [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+            "type": "Microsoft.Security/assessmentMetadata",
+            "properties": {
+              "displayName": "Install endpoint protection solution on virtual machine scale sets",
+              "policyDefinitionId": "/providers/Microsoft.Authorization/policyDefinitions/26a828e1-e88f-464e-bbb3-c134a282b9de",
+              "description": "Install an endpoint protection solution on your virtual machines scale sets, to protect them from threats and vulnerabilities.",
+              "remediationDescription": "To install an endpoint protection solution: 1.  <a href=\"https://docs.microsoft.com/azure/virtual-machine-scale-sets/virtual-machine-scale-sets-faq#how-do-i-turn-on-antimalware-in-my-virtual-machine-scale-set\">Follow the instructions in How do I turn on antimalware in my virtual machine scale set</a>",
+              "category": [ "Compute" ],
+              "requiredPricingBundle": [ "VirtualMachines" ],
+              "secureScoreWeight": 15,
+			  "assessmentType": "BuiltIn"
```
Comment (yungezz) [inline specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/examples/AssessmentsMetadata/ListAssessmentsMetadata_example.json]: indention

**Comment by tarosler** [comment]
@yungezz fixed all comments. please review once again.

Thanks,
Tal

File: specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/assessmentsMetadata.json [hunk 4]
```diff
@@ -0,0 +1,358 @@
...
+      "get": {
+        "x-ms-examples": {
+          "Get security assessment metadata for subscription": {
+            "$ref": "./examples/AssessmentsMetadata/GetAssessmentsMetadata_subscription_example.json"
+          }
+        },
+        "tags": [
+          "Assessments Metadata"
+        ],
+        "description": "Get metadata information on an assessment type in a specific subscription",
```
Comment (sanjaiganesh) [inline specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/assessmentsMetadata.json]: >ype in a specific subscriptio [](start = 67, length = 29)

I am unclear on the need of subscription specific apis..
How can assessment types be different for different subscriptions ?

Could you also clarify what is resource type 'assessment type 'refer to here ?
If it is /assessment/assessmentName resource type, how is these two resource types are different.. ?

Please schedule a meeting with api review board ..

Comment (tarosler) [reply specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/assessmentsMetadata.json]: Hi,

We are adding the notion of 'custom assessments' to ASC, giving the user the ability to create custom policy in Azure Policy, and load it as assessment to ASC. 

The custom assessments are defined in subscription scope, thus different subscriptions can have different assessments. This is the reason why we need metadata API in subscription scope.

the 'assessmentType' field represent whether the assessment is a BuilltIn assessment of ASC or a custom assessment defined by the user. This field is equivelent to the 'policyType' field in Azure Policy policy definitions API.

Thanks,
Tal.

**Review by ryansbenson — APPROVED** [review]
Approved from ARMs side

**Review by yungezz — APPROVED** [review]

**Pull request merged**